examples/gno.land/p/nt/simpledao/membstore.gno

```diff
@@ -0,0 +1,160 @@
+package simpledao
+
+import (
+	"errors"
+
+	"std"
+
+	"gno.land/p/demo/avl"
+	"gno.land/p/gov/dao"
+)
+
+var (
+	errAlreadyMember        = errors.New("address is already a member")
+	errMissingMember        = errors.New("address is not a member")
+	errInvalidAddressUpdate = errors.New("invalid address update")
+)
+
+type MembStoreOption func(*MembStore)
```

Comment:
How is this pattern future-proof if all the helpers can only be implemented locally (the struct only has private objects)?

I'm already skeptical about this pattern in Go, but I believe it's even worse in Gno. Opinions?

Reply:
It's not future proof by any means, you'd be dependent on the options provided by the `MembStore` implementation (for now only 1). The future proofing is that you can swap the govdao implementation at any point if you want a different API

The original DAO API had swap methods for both the `PropStore` and `MembStore`, so you could change the implementations without changing the DAO, but I decided against this in the end to not overcomplicate the API more